buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -184,6 +185,33 @@ class PerformanceTestPlugin : Plugin<Project> {
             // Never up-to-date since it reads data from the database.
             outputs.upToDateWhen { false }
         }
+
+        createPerformanceDefinitionJsonTask("writePerformanceScenarioDefinitions", performanceSourceSet, true)
+        createPerformanceDefinitionJsonTask("verifyPerformanceScenarioDefinitions", performanceSourceSet, false)
+    }
+
+    /**
+     * Scan performance test scenarios and:
+     * 1. Write to the scenario JSON file if write=true;
+     * 2. Only verify the scenario JSON file is correct if write=false.
+     */
+    private
+    fun Project.createPerformanceDefinitionJsonTask(name: String, performanceSourceSet: SourceSet, write: Boolean) {
+        val outputJson = project.rootProject.file(".teamcity/performance-tests-ci.json")
+        tasks.register<Test>(name) {
+            testClassesDirs = performanceSourceSet.output.classesDirs
+            classpath = performanceSourceSet.runtimeClasspath
+            maxParallelForks = 1
+            systemProperty("org.gradle.performance.scenario.json", outputJson.absolutePath)
+            systemProperty("org.gradle.performance.write.scenario.json", write)
+
+            if (write) {
+                outputs.cacheIf { true }
```

Comment:
The test task is already cacheable, so if you want to make it non-cacheable you need to do something in the else branch.